### DISCUSSION TITLE
Working with AnyOf for request bodies

### DISCUSSION THREAD

**Discussion by jaimyborgman** [dedoc/scramble#404] [Q&A]
Hey there,

First of all great start with this plugin, was thinking of making such a tool on my own but great to see that this one is maintained already. 

Now i'm getting into an issue; when defining multiple returns within the method the document is generator a nice "AnyOf" response; but how can we get that to work for request bodies; like in some cases you want to have two different kind of request bodies on the request

Is there any workaround or something missing in the documentation on how to get this to work or should i make an extension for it to improve?

Kind regards,
Jaimy

**Comment by romalytvynenko** [comment]
Hey @jaimyborgman 

Can you please share some code examples?

**Comment by jaimyborgman** [comment]
Sure:

```
/**
 * Register a new user
 *
 * @unauthenticated
 */
public function store(Request $request): Response
{
    $result = $this->validate($request, [
        'role' => ['required', Rule::in(['consumer', 'merchant'])],
        'email' => ['required', 'string', 'email', 'max:255', Rule::unique(User::class)],
        'locale' => ['required', 'string', Rule::in(Locale::toFrontend())],
        'company' => ['required_if:role,merchant', 'string'],
    ]);

    $role = Arr::get($result, 'role');
    $email = Arr::get($result, 'email');
    $locale = Arr::get($result, 'locale');

    $this->dispatch(new RegisterUser($role, $email, $locale));

    $this->authenticate($email);

    /** @status 204 */
    return $this->noContent();
}
```

i know that required_if isn't supported as of yet but was wondering if we can somehow "trick" the generator with annotations or something else as work-around to get Anyof as possibility for the request to

also there is an issue that the 204 is still giving an "string" as output which should not have any "response" at all; besides some controllers without having any validate step also gives a "{}" as request body which should not be there if there is no body needed at all - but i think there are some issues already open related to that
